### PR TITLE
(chore) - streamline setup

### PR DESCRIPTION
**Pull request by JoviDeCroock**
This provides our linting and precommit hooks

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/*
```
Comment (andyrichardson) [inline .npmignore]: Interesting way of only including files :+1:

File: .npmignore [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+!LICENSE
+!CHANGELOG.md
+!README.md
+!/dist/**/*.{d.ts,js,js.map}
```
Comment (andyrichardson) [inline .npmignore]:
```suggestion
!/dist
```
Just to be on the safe side, let's always include the full build output

File: tsconfig.json [hunk 3]
```diff
@@ -1,8 +1,21 @@
...
+    "rootDir": "./src",
+    "baseUrl": ".",
+    "outDir": "dist/cjs",
+    "lib": ["dom", "esnext"],
+    "jsx": "react",
+    "declaration": false,
+    "module": "es2015",
+    "moduleResolution": "node",
+    "target": "esnext",
+    "strict": false, // temp since this causes errors on used events.
```
Comment (andyrichardson) [inline tsconfig.json]: Let's fix this issue pre-merge

File: .editorconfig [hunk 4]
```diff
@@ -0,0 +1,9 @@
+root = true
```
Comment (andyrichardson) [inline .editorconfig]: Is this file needed if we're using prettier?

File: .eslintignore [hunk 5]
```diff
@@ -0,0 +1,2 @@
+dist
+examples/**/*
```
Comment (andyrichardson) [inline .eslintignore]: Skipping this
```suggestion

```

File: tsconfig.json (hunk 3 above)
Comment (andyrichardson) [reply tsconfig.json]: I've done a temporary cast. There's an issue about this here -https://github.com/Microsoft/TypeScript/issues/28357

File: .editorconfig (hunk 4 above)
Comment (JoviDeCroock) [reply .editorconfig]: We do it in other repo's so thought so

File: tsconfig.json [hunk 6]
```diff
@@ -1,8 +1,20 @@
 {
-  "include": ["src"],
   "compilerOptions": {
-    "declaration": true,
+    "esModuleInterop": true,
+    "noUnusedLocals": true,
```
Comment (andyrichardson) [inline tsconfig.json]: Let's leave this to eslint to handle
```suggestion
```

File: .editorconfig (hunk 4 above)
Comment (andyrichardson) [reply .editorconfig]: @JoviDeCroock fair enough :+1:
